api/hypershift/v1beta1/hostedcluster_types.go

```diff
@@ -2551,6 +2556,34 @@ type AESCBCSpec struct {
 	BackupKey *corev1.LocalObjectReference `json:"backupKey,omitempty"`
 }
 
+type PayloadArchType string
+
+const (
+	AMD64   PayloadArchType = "AMD64"
+	PPC64LE PayloadArchType = "PPC64LE"
+	S390X   PayloadArchType = "S390X"
+	ARM64   PayloadArchType = "ARM64"
+	Multi   PayloadArchType = "MULTI"
+)
+
+// ToPayloadArch converts a string to payloadArch.
+func ToPayloadArch(arch string) PayloadArchType {
+	switch strings.ToUpper(arch) {
```

Comment:
You can't rely on `strings.ToUpper` here, unless you also `strings.ToUpper` inside the case statements, but, realistically, this would be better to just be explicit, don't use a transform to avoid issues 

```
func ToPayloadArch(arch string) PayloadArchType {
  switch arch {
  case "amd64":
    return AMD64
  ...
  }
}
```